### DISCUSSION TITLE
[Feature Request] FileInput component

### DISCUSSION THREAD

**Discussion by notmedia** [heroui-inc/heroui#591] [Feature requests]
### Is your feature request related to a problem? Please describe.

Nextui missing file input component 🙏

### Describe the solution you'd like

Like any other file input component with pretty styling of nextui ❤️

**Comment by Albert-Jokelin** [comment]
What if we used an `upload` button with a text area beside it that displays the file uploaded. That should do the job

**Reply by notmedia** [reply]
Yep, something like this

<img width="386" alt="image" src="https://user-images.githubusercontent.com/12993018/177203321-86ef768f-6776-4410-98b8-2e0cb14172c4.png">

**Reply by Albert-Jokelin** [reply]
Looks cool. Do we have plans to make it, if so how do I contribute?

**Reply by notmedia** [reply]
I think first we need to get approve from @jrgarciadev

**Comment by niklasp** [comment]
I need it because <Input type="file"> is unusable atm

**Comment by dbradmit** [comment]
Any updates, alternative options, etc? I've been building with nextui for the first time recently and ran into this need.

**Comment by AirP0WeR** [comment]
Begin using NextUI, beautiful libruary. Found that it dont have simple fileinput or dropzone.

**Comment by AbidHasanPiash** [comment]
I create a custom container that looks like my other NextUi form field like this:
`
export default function NextUiContainer({children}) {
  return (
    <div 
        className="flex w-full items-center justify-between space-x-4 p-2 border-2 
        hover:border-gray-400 rounded-xl transition-colors duration-300">
        {children}
    </div>
  )
}
`
and then I use a simple input field for images like this:
`
<NextUiContainer>
            <input 
              type="file" 
              name="image" 
              onChange={handleFileChange} 
              className="block w-full text-sm text-slate-500
              file:mr-4 file:py-2 file:px-4 file:rounded-md
              file:border-0 file:text-sm file:font-semibold
              file:bg-pink-50 file:text-pink-700
              hover:file:bg-pink-100" />
</NextUiContainer>
`

**Comment by brunogarcia** [comment]
In the release May 1, React Aria has a new component called DropZone.
It seems an evolution of our old friend "input type file"

https://react-spectrum.adobe.com/react-spectrum/DropZone.html

**Comment by fillobotto** [comment]
Anyone?

**Comment by wingkwong** [comment]
There's already a [PR](https://github.com/nextui-org/nextui/pull/3832) in progress.
